python/dgl/utils.py

```diff
@@ -155,18 +187,23 @@ def get_items(self, index):
             # the provided index is not a slice
             tensor = self.tousertensor()
             index = index.tousertensor()
-            return Index(F.gather_row(tensor, index))
+            # TODO(Allen): Change F.gather_row to dgl operation
+            return Index(F.astype(F.gather_row(tensor, index),
+                                  F.data_type_dict[self.dtype]), self.dtype)
         elif self._slice_data is None:
             # the current index is not a slice but the provided is a slice
             tensor = self.tousertensor()
             index = index._slice_data
-            return Index(F.narrow_row(tensor, index.start, index.stop))
+            # TODO(Allen): Change F.narrow_row to dgl operation
```

Comment:
Why do you wish to use DGL operation?

Reply:
Some backend operation doesn't support int32 indexing therefore manually casting to long is needed in some cases. And these operation on 1D vector should be simple to implement, changing to DGL operation might be better?